### PR TITLE
Initial action implementation

### PR DESCRIPTION
**Pull request by OrBin**

### CODE REVIEW

File: action.yml [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    description: 'A personal API token to CircleCI'
+    required: true
+runs:
+  using: "composite"
+  steps:
+    - run: ${{ github.action_path }}/trigger_circleci_pipeline.sh ${{ inputs.triggered-workflow }}
+      shell: bash
+      env:
+          CIRCLECI_USER_PERSONAL_API_TOKEN: ${{ inputs.circleci-token }}
+          GITHUB_REPOSITORY: ${{ github.repository }}
```
Comment (sharonyogev) [inline action.yml:18]: don't you want to make these 2 also an input? or maybe input with defaults?

**Review by sharonyogev — CHANGES_REQUESTED** [review]

File: trigger_circleci_pipeline.sh [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+        status_response=$(
+            curl -sSX GET \
+                -H "Circle-Token: ${CIRCLECI_USER_PERSONAL_API_TOKEN}" \
+                "${CIRCLECI_API_BASE}/pipeline/${pipeline_id}" \
+            | jq -r '.state'
+        )
+
+        if [ "${status_response}" = "errored" ]; then
+            echo "Failed to create pipeline"
+            return 3
```
Comment (sharonyogev) [inline trigger_circleci_pipeline.sh]:
```suggestion
            exit 3
```

File: trigger_circleci_pipeline.sh [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+        fi
+
+        n=$((n+1))
+        sleep 3
+    done
+
+    if [ "${n}" = 5 ]; then
+      # Time out after 5 retries
+      echo "Timed out waiting for pipeline creation"
+      return 4
```
Comment (sharonyogev) [inline trigger_circleci_pipeline.sh]:
```suggestion
      exit 4
```

**Review by sharonyogev — CHANGES_REQUESTED** [review]

**Review by sharonyogev — APPROVED** [review]

**Comment by OrBin** [comment]
Tagged new version as `v1` and `v1.0`

**Pull request merged**